daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -807,7 +875,6 @@ class DirectOrBufferedDataOutputStream private[io] (var splitFrom: DirectOrBuffe
       }
     }
   }
-
 }
 
 final class BufferedBlobDataOutputStream private[io] (
```

Comment:
I question if this even needs to exist anymore. Seems to me that BlobDataOutputStream is really just a normal DirectOrBufferedDataOutputStrema except the ByteArrayOrFileOutputStreamWithGetBuf should be initialized with a File and never even create the ByteArrayOutputStream. Maybe we just need a special constructor/function that sets that all up instead of having this special class.

I think that would then get rid of some of the hacks where this overrides bufferingJOS and getJavaOutputStream.